Automattic/jetpack
# Site Migration: Add site migration endpoint

## Proposed changes:

* Add a `/wpcom/v2/sites/${site}/atomic-migration-status/migrate-guru-key` endpoint to WP.com sites from which we can fetch the migration key on the front end.
* Needs to be paired with D141148-code to work.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



Nope.

## Testing instructions:






* Apply this branch to your AT using Jetpack Beta plugin following instructions in https://github.com/Automattic/jetpack/pull/36253#issuecomment-1983872410
* Apply D141148-code to your sandbox and sandbox `public-api.wordpress.com`. 
* Apply this branch also on your sandbox by running `bin/jetpack-downloader test jetpack-mu-wpcom-plugin add/site-migration-endpoint-jetpack` (make sure this PR and D141148-code are running on your sandbox at the same time)
* Visit the developer console and fetch from `/wpcom/v2/sites/%your-AT-site%/atomic-migration-status/migrate-guru-key`
* You should see a 403 error
* On the AT site, install and activate the Migrate Guru plugin
* Fetch the endpoint from the developer console; the migration key field should be populated with the same migration key you'll find at `/wp-admin/admin.php?page=migrateguru`
* Try to fetch it again and verify that a 403 error is returned
